pkg/quarterdeck/middleware/auth.go

```diff
@@ -128,6 +130,24 @@ func GetClaims(c *gin.Context) (*tokens.Claims, error) {
 	return claims.(*tokens.Claims), nil
 }
 
+// ContextFromRequest creates a context from the gin request context, copying fields
+// that may be required for forwarded requests. This method should be called by
+// handlers which need to forward requests to other services and need to preserve data
+// from the original request such as the user's credentials.
+func ContextFromRequest(c *gin.Context) (ctx context.Context, err error) {
+	var req *http.Request
+	if req = c.Request; req == nil {
+		return nil, ErrNoRequest
```

Comment:
I would be very surprised if this ever happened … I regularly use `c.Request.Context()` without doing `nil` check on `c.Request` -- but this is very thorough!